src/Components/Components/src/Profiling/WebAssemblyComponentsProfiling.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using WebAssembly.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.Profiling
+{
+    // Later on, we will likely want to move this into the WebAssembly package. However it needs to
+    // be inlined into the Components package directly until we're ready to make the underlying
+    // ComponentsProfile abstraction into a public API. It's possible that this API will never become
+    // public, or that it will be replaced by something more standard for .NET, if it's possible to
+    // make that work performantly on WebAssembly.
+
+    internal class WebAssemblyComponentsProfiling : ComponentsProfiling
+    {
+        // When running on Blazor Server, the isCapturing flag will always be false, so the JS interop
+        // calls will not occur. This flag only gets set to true by a call from blazor.webassembly.js.
+        static bool IsCapturing = false;
+
+        public static void SetCapturing(bool isCapturing)
+        {
+            IsCapturing = isCapturing;
+        }
+
+        public override void Start(string? name)
+        {
+            if (IsCapturing)
+            {
+                InternalCalls.InvokeJSUnmarshalled<string, object, object, object>(
+                    out _, "_blazorProfileStart", name!, null!, null!);
```

Comment:
Why do we do `null!` here?

Reply:
The nullability hints on `InvokeJSUnmarshalled` say that the parameters have to be nonnull, but that's incorrect - they can be null (as they are here).

Thanks for pointing this out. I've (hopefully) fixed the underlying hints now and changed these to regular `null`.

Reply:
Great!